### ISSUE TITLE
Button group: check if we should use fieldset>legend instead of role="group" and aria-label

### ISSUE THREAD

**Issue by gfellerph** [swisspost/design-system#3023]
![image](https://github.com/swisspost/design-system/assets/1659006/61a19c7d-45ba-4326-b913-c921484346e3)

In our examples, the label is not visible, but in a real world application, a button group should have a visible label. For this case, I think it would be better to use a fieldset / legend construct (https://www.w3.org/WAI/tutorials/forms/grouping/#associating-related-controls-with-wai-aria).

A breaking change can be prevented by supporting the current structure as well as the fieldset > legend one. Possibly, the current structure could be a variant where it's not possible or necessary to show the label.

**Comment by gfellerph** [comment]
Closing as won't do (implementing this for v9)
